readthedocs/core/utils/__init__.py

```diff
@@ -165,8 +165,42 @@ def prepare_build(
         # External builds should be lower priority.
         options['priority'] = CELERY_LOW
 
+    skip_build = False
+    if commit:
+        skip_build = (
+            Build.objects
+            .filter(
+                project=project,
+                version=version,
+                commit=commit,
+            ).exclude(
+                state=BUILD_STATE_FINISHED,
+            ).exists()
+        )
+    else:
+        skip_build = Build.objects.filter(
+            project=project,
+            version=version,
+            state=BUILD_STATE_TRIGGERED,
+        ).count() > 1
+    if skip_build:
+        # TODO: we could mark the old build as duplicated, however we reset our
+        # position in the queue and go back to the end of it --penalization
```

Comment:
I don't see where we're changing the queue position?

Reply:
We are queuing the new build at the end of the queue, but we already had a build closer to be executed. If we "cancel" the build that was closer to be executed because the new build is going to replace it, we loose that position in the queue.